### ISSUE TITLE
rebuild-pods dont get deleted

### ISSUE THREAD

**Issue by mahnouman** [k8-proxy/go-k8s-controller#35]
@Lestat06 commented on [Tue Jul 06 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123)

the rebuild-pod that complete the task should be deleted immediately which is not, hence the rebuild pod number increases.

you can replicate by running the instance in f2f with big amount of files scenarios like govUk(aprox 10k files).
u will se that after 6k files the the rebuild process struggles. because of the pods.

![image](https://user-images.githubusercontent.com/6232283/124597499-de70e000-de63-11eb-90e0-df3885c225ae.png)
![image (1)](https://user-images.githubusercontent.com/6232283/124597529-e597ee00-de63-11eb-8ade-887bd5aa5121.png)
![image](https://user-images.githubusercontent.com/6232283/124597531-e6308480-de63-11eb-9dde-275509d5caf3.png)

---

@GINAGC commented on [Tue Jul 06 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-874891000)

@mahnouman who can review this one?

---

@mahnouman commented on [Wed Jul 07 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-875326466)

@GINAGC assigned to @haitham911 

---

@mahnouman commented on [Wed Jul 07 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-875326831)

NOTE: possible duplicate of https://github.com/k8-proxy/go-k8s-infra/issues/121

---

@mahnouman commented on [Wed Jul 07 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-875329366)

NOTE: Possible cause is the lately introduced Burstable configuration

---

@mahnouman commented on [Wed Jul 07 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-875332580)

NOTE: possible duplicate of https://github.com/k8-proxy/go-k8s-infra/issues/116

---

@himanshu2464 commented on [Wed Jul 07 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-875466130)

and we should limit the pod's counts like for example if we are running the 50 threads, then only 50 rebuild pods should be there, not more than that.

---

@GINAGC commented on [Thu Jul 08 2021](https://github.com/k8-proxy/go-k8s-infra/issues/123#issuecomment-876378149)

Workaround:
when this pods failed to launch container due to ip not able to assign and IP pool get full perform following thing to make the k3s back and responsive.

- run the following with root user by sudo su
- cd /var/lib/cni/networks 
-  systemctl stop k3s
- systemctl stop containerd
- mv cbr0/ cbr0.bak
- mkdir cbr0
- systemctl start containerd
- systemctl start k3s
- mkdir -p ~/.kube && sudo install -T /etc/rancher/k3s/k3s.yaml ~/.kube/config -m 600 -o $USER

and your all pods will be up and running .
and cluster will be responsive back

**Comment by Lestat06** [comment]
this is still happening. now at 5.5k files. and also the script worked for a little bit then the issue happened again

![image (1)](https://user-images.githubusercontent.com/6232283/124997893-bb088980-e04b-11eb-997c-6b4d643743c2.png)
![image](https://user-images.githubusercontent.com/6232283/124997895-bcd24d00-e04b-11eb-9c89-00a520115396.png)
![image](https://user-images.githubusercontent.com/6232283/124997951-d378a400-e04b-11eb-8135-6cbaf0832e8f.png)

**Comment by GINAGC** [comment]
Blocked by https://github.com/k8-proxy/go-k8s-infra/issues/125

**Comment by Lestat06** [comment]
Due to below incident I cant test this one from f2f since is in develop branch FYI @GINAGC 
https://github.com/filetrust/cdr-plugin-folder-to-folder-bugs/issues/71

**Comment by Lestat06** [comment]
@GINAGC we can move this to release since is not an issue anymore
